backend/dataall/modules/dataset_sharing/services/share_notification_service.py

```diff
@@ -1,152 +1,140 @@
+            - dataset.stewards
+            - share.groupUri
+        """
         notifications = []
-        targeted_users = ShareNotificationService._get_share_object_targeted_users(
-            session, dataset, share
-        )
-        for user in targeted_users:
+        for user in self.notification_target_users:
+            log.info(f"Creating notification for {user}, msg {msg}")
             notifications.append(
-                Notification.create(
-                    session=session,
+                NotificationRepository.create_notification(
+                    session=self.session,
                     username=user,
-                    notification_type=NotificationType.DATASET_VERSION,
-                    target_uri=f'{share.shareUri}|{dataset.datasetUri}',
-                    message=f'New data (at {s3_prefix}) is available from dataset {dataset.datasetUri} '
-                            f'shared by owner {dataset.owner}',
+                    notification_type=notification_type,
+                    target_uri=f'{self.share.shareUri}|{self.dataset.datasetUri}',
+                    message=msg,
                 )
             )
-            session.add_all(notifications)
+        self.session.add_all(notifications)
         return notifications
 
-    @staticmethod
-    def _get_share_object_targeted_users(session, dataset, share):
-        targeted_users = ShareNotificationService._get_dataset_stewards(dataset)
-        targeted_users.append(dataset.owner)
-        targeted_users.append(share.owner)
-        return targeted_users
-
-    @staticmethod
-    def _get_dataset_stewards(dataset):
-        stewards = list()
-        stewards.append(dataset.SamlAdminGroupName)
-        stewards.append(dataset.stewards)
-        return stewards
-
-    @staticmethod
-    def create_notification_task(session, email_id, dataset, share, subject, msg):
-        share_notification_config = config.get_property('core.share_notifications')
-
-        notification_recipient_groups_list = [dataset.SamlAdminGroupName, dataset.stewards]
+    def _create_notification_task(self, subject, msg):
+        """
+        At the moment just for notification_config_type = email, but designed for additional notification types
+        Emails sent to:
+            - dataset.SamlAdminGroupName
+            - dataset.stewards
+            - share.owner (person that opened the request) OR share.groupUri (if group_notifications=true)
+        """
+        share_notification_config = config.get_property('modules.datasets.share_notifications')
+
+        notification_recipient_groups_list = [self.dataset.SamlAdminGroupName, self.dataset.stewards]
         notification_recipient_email_ids = []
 
         if share_notification_config['email']['features']['group_notifications'] == True:
-            notification_recipient_groups_list.append(share.groupUri)
+            notification_recipient_groups_list.append(self.share.groupUri)
```

Comment:
Should we only check if the `group_notifications` piece of this feature is true/false after we determine the `email` feature is active at all?

Meaning if I just set `modules.datasets.share_notifications.email.active = False` and specify nothing for group_notifications I think this will fail

Reply:
Added a couple of extra checks! Thank you

Reply:
And as for the problem with the notifications and who sees them as the output of listNotifications(). That is an issue that was there originally. Let me see if I can fix it as part of this PR @noah-paige

Reply:
I have done the changes +  some clean-up of the notifications queries. I noticed that there are 2 queries that are currently unused in the frontend (`countDeletedNotifications`, `countReadNotifications`). Should we remove this unused code as well? For the moment I am leaving a comment

Reply:
This will be handled in future enhancements to notifications